### ISSUE TITLE
The search doesn't work if language="es" (spanish)

### ISSUE THREAD

**Issue by meisterchecho** [sphinx-doc/sphinx#2581]
hi guys, search words using language="es" in the config file doesn't work. This is my output

Running Sphinx v1.4.1
loading translations [es]... done
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 21 source files that are out of date
updating environment: [config changed] 21 added, 0 changed, 0 removed
reading sources... [100%] 
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] unificado
generating indices... genindex
writing additional pages... search
copying images... [100%] 
copying static files... done
copying extra files... done
dumping search index in Spanish (code: es) ... done
dumping object inventory... done
build succeeded.

But using language=en works so far.
my env.
alabaster==0.7.8
Babel==2.3.4
colorama==0.3.7
docutils==0.12
imagesize==0.7.1
Jinja2==2.8
MarkupSafe==0.23
Pygments==2.1.3
pytz==2016.4
six==1.10.0
snowballstemmer==1.2.1
Sphinx==1.4.1
sphinx-rtd-theme==0.1.9

thank you in advance

**Comment by tk0miya** [comment]
In my short investigation:

With no language settings, the `searchindex.js` has lowercased terms:

```
Search.setIndex({envversion:48,filenames:["index"],objects:{},objnames:{},objtypes:{},terms:{content:0,index:0,modul:0,page:0,search:0},titles:["Welcome to test&#8217;s documentation!"],titleterms:{document:0,indic:0,tabl:0,test:0,welcom:0}})
```

But, with `language='es'`, the index file has non lowercased terms:

```
Search.setIndex({envversion:48,filenames:["index"],objects:{},objnames:{},objtypes:{},terms:{"\u00cdndic":0,Busqued:0,Contents:0,Modul:0,Pagin:0},titles:["Welcome to test&#8217;s documentation!"],titleterms:{Indic:0,Welcom:0,and:0,documentation:0,tabl:0,test:0}})
```

It seems the terms are not lowercased in Spanish environment.

@shibukawa Can I make them lowercase before storing into indices?
It seems that the terms are not lowercased in some languages.

**Comment by tk0miya** [comment]
I discussed with shibukawa and fixed this at 9226513.
Thank you for reporting.

**Comment by meisterchecho** [comment]
thank you! great job! :+1: for sphinx.
